core/src/main/scala/org/apache/spark/shuffle/ShuffleManager.scala

```diff
@@ -54,6 +54,19 @@ private[spark] trait ShuffleManager {
       context: TaskContext,
       metrics: ShuffleReadMetricsReporter): ShuffleReader[K, C]
 
+  /**
+   * Get a reader for a range of reduce partitions (startPartition to endPartition-1, inclusive) to
+   * read from mapId.
+   * Called on executors by reduce tasks.
+   */
+  def getMapReader[K, C](
```

Comment:
instead of creating a new API, how about we set the map id in `TaskContext.getLocalProperties`, and still call the existing `getReader` method?

Reply:
May be the `getMapReader`  can distinguish whether based on the mapId ?